backend/server/helpers/userHelper.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @file Defines a helper for common user actions
+ */
+
+ const admin = require('firebase-admin');
+ const winston = require('winston');
+
+ /**
+  * Looks up a firebase user by their email
+  * @param {string} email the email of the user to lookup
+  * @returns {Promise} A promise that contains a Firebase DataSnapshot with the relevant user
+  */
+ const getFirebaseUserByEmail = (email) => {
+   winston.info('userHelper.getFirebaseUserByEmail - looking up user with email %s', email);
+
+   try {
+     const usersRef = admin.database().ref('/users');
+     return usersRef.orderByChild('email').equalTo(email).limitToFirst(1)
+      .once('child_added').catch((error) => {
+        winston.error('userHelper.getFirebaseUserByEmail - Error looking up user by email %s. Error message: %s', email, error.message);
+      });
+    }
+    catch (error) {
+      winston.error('userHelper.getFirebaseUserByEmail - Error looking up user by email %s. Error message: %s', email, error.message);
+    }
+ };
+
+ /**
+  * Adds a user specified by email to the canvas designated by canvasId
+  * @param {string} email the email of the user to add the the canvas
+  * @param {string} canvasId the canvas id of the canvas to add the user to
+  * @returns {void}
+  */
+ const addUserToCanvasByEmail = (email, canvasId) => {
+  winston.info('userHelper.addUserToCanvasByEmail - adding user %s to canvas %s', email, canvasId);
+
+  try {
```

Comment:
Would it be possible to break up this try block into specific parts where we might fail so there isn't a giant block of code in the try.